lib/ansible/modules/cloud/vmware/vmware_datastore_upload.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2015, Dag Wieers (@dagwieers) <dag@wieers.com>
+# Copyright: (c) 2018, James E. King III (@jeking3) <jking@apache.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+# TODO
+# - make unit tests
+# - test what happens when the first host is down, goes to second
+# - test what happens if all hosts are down
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_datastore_upload
+short_description: Upload a local file to a VMware datastore.
+description:
+    - Upload a local file to a VMware datastore with the option to
+      skip the upload if the file already exists and is the same.
+author:
+    - "James E. King III (@jeking3) <jking@apache.org>"
+version_added: 2.8
+notes:
+    - Tested on vSphere 6.7
+requirements:
+    - python >= 2.6
+    - PyVmomi
+    - requests >= 2.18.4
+options:
+    datacenter:
+      description:
+      - Name of the datacenter containing the datastore.
+      - If not specified, and more than one datastore with the
+        same name exists, the action will fail.
+      required: no
```

Comment:
Could you add `aliases: [ 'datacenter_name' ]`? Lots of other modules use datacenter_name and don't have an alias defined..

Reply:
Sure, someone should probably make a PR to standardize this across the modules that are there.